cmd/update/volume_move.go

```diff
@@ -117,71 +121,87 @@ func (c *volumeMoveCommand) runWithCtx(ctx context.Context, cmd *cobra.Command,
-		DeploymentIDs: deploymentsOnSourceNode,
-	}
-	if c.useCAS() {
-		params.CASVersion = vol.Version
-	}
-	err = c.client.SetPreferredEvictionCandidates(ctx, nsID.String(), vol.ID.String(), params)
-	if err != nil {
-		return fmt.Errorf("failed to set preferred eviction candidates: %s", err)
-	}
+	if !isMovingMaster {
+		// mark the replica on the [source] node as preferred for eviction
+		// the master is evicted directly through an api call instead
 
-	c.display.TextMessage(ctx, c.writer, "deployments on [source] node marked as preferred for eviction")
+		params := &apiclient.SetPreferredEvictionCandidatesRequestParams{
+			DeploymentIDs: deploymentsOnSourceNode,
+		}
+		if c.useCAS() {
+			params.CASVersion = vol.Version
+		}
+		err = c.client.SetPreferredEvictionCandidates(ctxLongTimeout, nsID.String(), vol.ID.String(), params)
+		if err != nil {
+			return fmt.Errorf("failed to set preferred eviction candidates: %s", err)
+		}
+
+		c.display.TextMessage(ctxLongTimeout, c.writer, fmt.Sprintf("deployments on [source] node %s marked as preferred for eviction", c.fromNode))
+	}
 
 	// wait for the new replica on [destination] node to sync
-	err = c.waitForReplicaOnNodeToSync(ctx, nsID)
+	err = c.waitForReplicaOnNodeToSync(ctxLongTimeout, nsID)
 	if err != nil {
 		return err
 	}
 
-	c.display.TextMessage(ctx, c.writer, "replica on [destination] node finished syncing")
+	c.display.TextMessage(ctxLongTimeout, c.writer, fmt.Sprintf("replica on [destination] node %s finished syncing", c.toNode))
 
 	if !isMovingMaster {
 		return nil
 	}
 
-	// find the deployment ID of the newly created one from the call above
+	// from here on, we are attempting to move a master
+
+	// find the deployment IDs of the newly created replica and the master being moved
 	var newReplicaID id.Deployment
 	for _, rep := range vol.Replicas {
 		if rep.Node.String() == c.toNode {
 			newReplicaID = rep.ID
 		}
 	}
+	oldMasterID := vol.Master.ID
 
-	err = c.client.AttemptPromotion(ctx, string(nsID), vol.ID.String(), newReplicaID.String())
+	err = c.client.AttemptPromotion(ctxLongTimeout, string(nsID), vol.ID.String(), newReplicaID.String())
 	if err != nil {
 		return err
 	}
 
-	c.display.TextMessage(ctx, c.writer, fmt.Sprintf("replica on [destination] node %s attempting promotion to master", c.toNode))
+	c.display.TextMessage(ctxLongTimeout, c.writer, fmt.Sprintf("replica on [destination] node %s attempting promotion to master", c.toNode))
 
-	err = c.waitForMasterPromote(ctx, nsID)
+	err = c.waitForMasterPromote(ctxLongTimeout, nsID)
 	if err != nil {
 		return err
 	}
 
-	c.display.TextMessage(ctx, c.writer, fmt.Sprintf("node on [destination] node %s successfully promoted to master", c.toNode))
+	c.display.TextMessage(ctxLongTimeout, c.writer, fmt.Sprintf("replica on [destination] node %s successfully promoted to master", c.toNode))
+
+	_ = c.client.EvictReplica(ctxLongTimeout, nsID.String(), vol.ID.String(), oldMasterID.String())
```

Comment:
It looks like we're ignoring this error here, but then checking it in the next line

Reply:
Nice spot, it was indeed being ignored by mistake